File: frontend/src/types/bom.ts
/**
 * BOM（物料清单）相关类型定义
 * 定义BOM管理模块的数据类型和接口
 */

// 物料明细项
export interface MaterialItem {
  materialID: string     // 物料唯一ID
  materialName: string   // 物料名称
  spec: string           // 规格型号
  quantity: number       // 数量
  supplierID: string     // 供应商ID
}

// BOM信息
export interface BOM {
  bomID: string              // BOM唯一ID
  bomType: string            // BOM类型：研发BOM/生产BOM
  productModel: string       // 对应车型
  partBatchNo: string        // 零部件批次号
  version: string            // BOM版本号
  creator: string            // 创建人
  createTime: string         // 创建时间
  status: string             // 状态：草稿/已发布/已变更/已作废
  materialList: MaterialItem[] // 物料明细列表
}

// BOM比较请求
export interface BOMCompareRequest {
  productionBOMID: string
  rdBOMID: string
}

// BOM变更请求
export interface BOMChangeRequest {
  bomID: string
  changeID: string
  changeReason: string
  oldVersion: string
  newVersion: string
  changeTime: string
}

// 创建BOM请求
export interface CreateBOMRequest {
  bomID: string
  bomType: string
  productModel: string
  partBatchNo: string
  version: string
  creator: string
  status?: string
  materialList: MaterialItem[]
}

// 更新BOM请求
export interface UpdateBOMRequest extends CreateBOMRequest {}

File: backend/model/bom.go
package model

type BOM struct {
	BOMID             string             `json:"bomID"`
	BOMType           string             `json:"bomType"`
	ProductModel      string             `json:"productModel"`
	PartBatchNo       string             `json:"partBatchNo"`
	Version           string             `json:"version"`
	Creator           string             `json:"creator"`
	CreateTime        string             `json:"createTime"`
	Status            string             `json:"status"`
	MaterialList      []MaterialItem     `json:"materialList"`
	RDBOMFileInfo     *BOMFileInfo       `json:"rdBOMFileInfo,omitempty"`
	ProductionBOMInfo *ProductionBOMInfo `json:"productionBOMInfo,omitempty"`
	ChangeRecords     []BOMChangeRecord  `json:"changeRecords,omitempty"`
}

type BOMDTO struct {
	BOMID        string         `json:"bomID" binding:"required"`
	BOMType      string         `json:"bomType" binding:"required"`
	ProductModel string         `json:"productModel" binding:"required"`
	PartBatchNo  string         `json:"partBatchNo" binding:"required"`
	Version      string         `json:"version" binding:"required"`
	Creator      string         `json:"creator" binding:"required"`
	CreateTime   string         `json:"createTime"`
	Status       string         `json:"status"`
	MaterialList []MaterialItem `json:"materialList" binding:"required"`
}

type MaterialItem struct {
	MaterialID   string `json:"materialID"`
	MaterialName string `json:"materialName"`
	Spec         string `json:"spec"`
	Quantity     int    `json:"quantity"`
	SupplierID   string `json:"supplierID"`
}

type BOMFileInfo struct {
	FileStorageID  string `json:"fileStorageID"`
	FilePath       string `json:"filePath"`
	FileHash       string `json:"fileHash"`
	FileType       string `json:"fileType"`
	FileSize       int64  `json:"fileSize"`
	UploadTime     string `json:"uploadTime"`
	DatabaseSource string `json:"databaseSource"`
}

type ProductionBOMInfo struct {
	RDOMReferenceID string `json:"rdBOMReferenceID"`
	CompareResult   string `json:"compareResult"`
	CompareDetails  string `json:"compareDetails"`
	Verifier        string `json:"verifier"`
	VerifyTime      string `json:"verifyTime"`
}

type BOMChangeRecord struct {
	ChangeID     string      `json:"changeID"`
	ChangeReason string      `json:"changeReason"`
	OldVersion   string      `json:"oldVersion"`
	NewVersion   string      `json:"newVersion"`
	AuditFlow    []AuditNode `json:"auditFlow"`
	ChangeTime   string      `json:"changeTime"`
}

type AuditNode struct {
	Auditor     string `json:"auditor"`
	AuditRole   string `json:"auditRole"`
	AuditResult string `json:"auditResult"`
	AuditTime   string `json:"auditTime"`
	Remark      string `json:"remark,omitempty"`
}
